### PR TITLE
exit to gamemode menu

### PR DESCRIPTION
**Pull request by youbitchoc**
STORY SPECIFIC STUFF (or at least shouldn't apply to meadow)
- [x] reset storylobbydata/storyclientsettings when switching campaigns
- [x] kick players to slugcat selection menu when host exits out of game session
  - don't think we should allow host to leave temporarily
- [ ] kick players to lobby when host quits

### CODE REVIEW

**Comment by 6fears7** [comment]
Is there a significant barrier to maintaining a headless session in story, assuming at least one player chose to overwrite their save with the hosts? Thinking we could leverage PickNewOwner for this unless there are more challenges I'm not thinking of

**Comment by youbitchoc** [comment]
if host wants to hand off then i think it should be an explicit choice.

ownership transfer shouldn't be allowed mid-game because client saves won't be consistent, once game session ends then we can sync host's savestate (see #289)

can probably reuse the playerlist in lobby for the handing off ui

maybe host could define a next-in-succession? so if host leaves midgame everyone still gets kicked to lobby but can restart the cycle as if nothing happened

**Comment by 6fears7** [comment]
Hmm ok let's go with ending game session for now. Handoff could be useful in the event host crashes and we need to determine a next best fit for the game to continue but that's a nice to have versus a stability feature which is the current aim of this PR. 

Full steam ahead 🛥️

File: GameModes/StoryGameMode.cs [hunk 1]
```diff
@@ -75,7 +75,6 @@ public override void LobbyReadyCheck()
             if (lobby.isOwner)
             {
                 RainMeadow.Debug("Host LobbyReadyCheck - started game");
-                currentCampaign = storyClientSettings.playingAs;
```
[the remark this answers is not in the document]
Comment (youbitchoc) [reply GameModes/StoryGameMode.cs:78]: it's already set on host slugcat select

if ownership changes then campaign will change to scug of new host, which breaks the game pretty badly

**Comment by youbitchoc** [comment]
need to test this in arena

**Comment by youbitchoc** [comment]
works fine in arena

**Pull request merged**